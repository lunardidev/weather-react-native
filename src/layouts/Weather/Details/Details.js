import React from 'react';

import {
  Container,
  TabContainer,
  TabSubText,
  TabTextContainer,
  TabTitleText,
  RefreshButton,
} from './styles';

const Details = (props) => {
  const { humidity, windSpeedy, sunrise, sunset, updateData } = props;

  return (
    <Container>
      <TabContainer>
        <TabTextContainer>
          <TabTitleText>Humidade: </TabTitleText>
          <TabSubText>{humidity}</TabSubText>
        </TabTextContainer>
        <TabTextContainer>
          <TabTitleText>Velocidade do vento: </TabTitleText>
          <TabSubText>{windSpeedy}</TabSubText>
        </TabTextContainer>
        <TabTextContainer>
          <TabTitleText>Nascer do Sol: </TabTitleText>
          <TabSubText>{sunrise}</TabSubText>
        </TabTextContainer>
        <TabTextContainer>
          <TabTitleText>PÃ´r do Sol: </TabTitleText>
          <TabSubText>{sunset}</TabSubText>
        </TabTextContainer>
        <RefreshButton title="Atualizar" onPress={() => updateData()} />
      </TabContainer>
    </Container>
  );
};

export default Details;
