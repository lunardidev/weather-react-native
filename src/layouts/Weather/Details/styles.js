import styled from 'styled-components/native';
import { View } from 'react-native';

export const Container = styled(View)``;

export const TabContainer = styled.ScrollView.attrs({
  horizontal: false,
  showsHorizontalScrollIndicator: false,
  contentContainerStyle: { paddingLeft: 0, paddingRight: 0 },
})`
  height: 100%;
  padding: 20px;
  padding-top: 40px;
`;

export const TabTextContainer = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 10px;
`;

export const TabTitleText = styled.Text`
  font-size: 16px;
  font-weight: normal;
  color: #232526;
  flex: 2;
  font-weight: 600;
`;

export const TabSubText = styled.Text`
  display: flex;
  flex: 1;
  flex-direction: row;
  font-size: 16px;
  font-weight: normal;
  color: #112d32;
  font-weight: 400;
`;

export const RefreshButton = styled.Button``;
