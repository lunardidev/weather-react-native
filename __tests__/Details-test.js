import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../src/layouts/Weather/Details/Details';

test('should render the weather details correctly', () => {
  const element = renderer
    .create(
      <Details
        humidity={10}
        windSpeedy="200km/h"
        sunrise="5.45pm"
        sunset="6.25am"
        updateData={() => jest.fn()}
      />
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
