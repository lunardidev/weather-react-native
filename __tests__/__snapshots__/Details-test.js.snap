// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the weather details correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    horizontal={false}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "height": "100%",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 40,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#232526",
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Humidade: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#112d32",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          10
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#232526",
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Velocidade do vento: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#112d32",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          200km/h
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#232526",
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          Nascer do Sol: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#112d32",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          5.45pm
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#232526",
                "flexBasis": 0,
                "flexGrow": 2,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "600",
              },
            ]
          }
        >
          PÃ´r do Sol: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#112d32",
                "display": "flex",
                "flexBasis": 0,
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          6.25am
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Atualizar
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
